packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -297,12 +304,24 @@ export class StateMachine extends StateMachineBase {
       }));
     }
 
+    let tracingConfiguration: CfnStateMachine.TracingConfigurationProperty | undefined;
+    if (props.tracingEnabled) {
+      tracingConfiguration = {
+        enabled: true,
+      };
+      this.addToRolePolicy(new iam.PolicyStatement({
+        actions: ['xray:PutTraceSegments', 'xray:PutTelemetryRecords'],
+        resources: ['*'],
+      }));
+    }
```

Comment:
i'd like to move this stuff out of the constructor to keep it light. It was probably not a good idea to have logging configuration in there but let's not proliferate that.

how about moving this into a method called `setupXray()` or `setupTracing()` which will return the configuration as well as make the call to add role policy

Reply:
This was just trying to stay consistent with the preceding block for logging configuration. I'll move them both to simplify the constructor.